File: .claude/hooks/subagent-start-tts.mjs
#!/usr/bin/env node
// @hookstack subagent-start-tts-announce
// Annonce le démarrage d'un sous-agent par TTS (SubagentStart)
import { execSync } from 'node:child_process';
import { fileURLToPath } from 'node:url';

function defaultExec(cmd) {
  execSync(cmd, { timeout: 10_000, stdio: 'ignore', shell: true });
}

export function run({ exec = defaultExec, platform = process.platform } = {}) {
  const text = 'Sous-agent démarré';
  try {
    if (platform === 'darwin') exec(`say "${text}"`);
    else exec(`espeak "${text}" 2>/dev/null || spd-say "${text}"`);
  } catch {}
  return text;
}

/* v8 ignore next 3 */
if (process.argv[1] === fileURLToPath(import.meta.url)) {
  run();
}

File: .claude/hooks/subagent-start-tts.py
#!/usr/bin/env python3
# @hookstack subagent-start-tts-announce
"""Announces subagent startup by TTS (SubagentStart)."""
import subprocess
import sys


def _default_exec(cmd):
    subprocess.run(
        cmd, shell=True, timeout=10, check=True, stdout=subprocess.DEVNULL, stderr=subprocess.DEVNULL
    )


def run(*, exec_cmd=None, platform=None):
    if exec_cmd is None:
        exec_cmd = _default_exec
    if platform is None:
        platform = sys.platform

    text = "Sous-agent démarré"
    try:
        if platform == "darwin":
            exec_cmd(f'say "{text}"')
        else:
            exec_cmd(f'espeak "{text}" 2>/dev/null || spd-say "{text}"')
    except Exception:
        pass
    return text


if __name__ == "__main__":
    run()
